src/rdf4cpp/rdf/datatypes/registry/util/CharConvExt.hpp

```diff
@@ -124,41 +131,46 @@ std::string to_chars(F const value) noexcept {
     static constexpr size_t buf_sz = 5 + std::numeric_limits<F>::max_digits10 + std::max(2ul, detail::log10ceil(std::numeric_limits<F>::max_exponent10));
     std::array<char, buf_sz> buf;
 
-    std::to_chars_result res = [&]() {
-        [[maybe_unused]] F integral;
-        F const fraction = std::modf(value, &integral);
-
-        if (fraction == 0) {
-            // include one zero after dot
-            return std::to_chars(buf.data(), buf.data() + buf.size(), value, std::chars_format::scientific, 1);
-        } else {
-            // automatic precision guarantees round-trip-safety (https://en.cppreference.com/w/cpp/utility/to_chars)
-            return std::to_chars(buf.data(), buf.data() + buf.size(), value, std::chars_format::scientific);
-        }
-    }();
+    std::to_chars_result res = std::to_chars(buf.data(), buf.data() + buf.size(), value, std::chars_format::scientific);
```

Comment:
I could not find a detailed specification of the format `std::chars_format::scientific`. 

How do we know that it adheres to these two rules: 

> - If SV has an absolute value that is greater than or equal to 0.000001 (one millionth) and less than 1000000 (one million), then the value is converted to an xs:decimal and the resulting xs:decimal is converted to an xs:string according to the rules above, as though using an implementation of xs:decimal that imposes no limits on the totalDigits or fractionDigits facets. 
> - n other cases, the result consists of a mantissa, which has the lexical form of an xs:decimal, followed by the letter "E", followed by an exponent which has the lexical form of an xs:integer.

from Casting to [xs:string and xs:untypedAtomic](https://www.w3.org/TR/xpath-functions/#casting-to-string)

Reply:
It is specified in terms of `printf` so `std::chars_format::scientific` maps to `printf`'s `e` modifier.

https://en.cppreference.com/w/cpp/utility/to_chars At 4) "[..]  but the conversion specified for the as-if printf is [..] `e` if fmt is `std::chars_format::scientific`"

https://en.cppreference.com/w/c/io/fprintf in the table:  "converts floating-point number to the hexadecimal exponent notation. [..] For the `e` conversion style `[-]d.ddde±dd` is used."

Reply:
Have you checked that it prints more than 6 digits after the decimal point? The fprintf definition suggests that precision is limited to 6 digits. From what I could seen xsd there is no such limit and I am quite sure that we need more than 6 decimal digits to represent the precision of double

Reply:
https://en.cppreference.com/w/cpp/utility/to_chars

3) "[..] parsing the representation using the corresponding `std::from_chars` function recovers value exactly." 
4) "same as (3), [but specified format]"

Would suggest that it has to, but I'll check quickly.

Reply:
Yes, behaves as expected. Added small test case